### PR TITLE
Add source_pr_number input for use with workflow_dispatch events.

### LINKED ISSUE #433: Workflow dispatch support
**Is your feature request related to a problem? Please describe.**

I'd like to be able to manually trigger the workflow with the PR number as an input.

I have a couple of use cases for this:

- I make changes to the GHA inputs and want it to execute against a PR that's already been merged
- The PR was merged before the workflow existed
- The PR was missing the labels the workflow selected for when it was merged
- The workflow failed (retrying might work, though if it failed and needed updates to the inputs, then this won't work, since retrying reuses the workflow at the time the job was executed).

Based on the code here: https://github.com/korthout/backport-action/blob/main/src/github.ts#L57-L65 and some trial/error, it doesn't seem to work with workflow dispatch events.

**Describe the solution you'd like**
An input variable to specify the PR number to run against, and if set, it takes precedence over the value in the  payload.

**Describe alternatives you've considered**

It could directly check the payload for workload_dispatch rather than accepting new inputs for the PR directly, however I that's less flexible and requires the workflow to be written a particular way.

**Additional context**

I tried with workflow dispatch already and got this error:
```
Run chancez/backport-action@pr/chancez/static_labels
Retrieve pull request data for #undefined
Check whether pull request undefined is merged
Create comment: Only merged pull requests can be backported.
HttpError: Not Found
    at /home/runner/work/_actions/chancez/backport-action/pr/chancez/static_labels/node_modules/@octokit/request/dist-node/index.js:112:1
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### PR DESCRIPTION
**Pull request by chancez**
Fixes #433

I've tested this in the repo I'm using this with and its been working well for manually triggering on some PRs that we merged prior to introducing the GHA.

### CODE REVIEW

**Review by korthout — COMMENTED** [review]
Thanks @chancez for this great contribution!

I finally had some time to look at open pull requests here. Sorry for the long wait 🙇 

I really like the feature and will gladly support it. There's just one question remaining for me: why should this only be supported for `workflow_dispatch`?

EDIT: I've taken the liberty to remove this restriction already, but if you see a reason for this we can take away these commits again. Let me know what you think.

**Comment by chancez** [comment]
> Thanks @chancez for this great contribution!
> 
> I finally had some time to look at open pull requests here. Sorry for the long wait 🙇
> 
> I really like the feature and will gladly support it. There's just one question remaining for me: why should this only be supported for `workflow_dispatch`?
> 
> EDIT: I've taken the liberty to remove this restriction already, but if you see a reason for this we can take away these commits again. Let me know what you think.

I don't see any big reason, I think I just was being cautious, because I think I remember seeing the GHA already checking the event type in another case, so I think I just copied it. I can't remember exactly though.

File: src/backport.ts [hunk 1]
```diff
@@ -104,7 +105,19 @@ export class Backport {
 
       if (repo === undefined) throw new Error("No repository defined!");
 
-      const pull_number = this.github.getPullNumber();
+      if (
+        this.config.source_pr_number !== undefined &&
+        this.github.getEventName() !== "workflow_dispatch"
+      ) {
+        throw new Error(
+          "source_pr_number can only be specified for workflow_dispatch events!",
```
[the remark this answers is not in the document]
Comment (chancez) [reply src/backport.ts]: I think this is fine. I believe I was just being cautious.

**Review by korthout — APPROVED** [review]
👍 LGTM

**Comment by korthout** [comment]
Released in [`v3.1.0`](https://github.com/korthout/backport-action/releases/tag/v3.1.0)

**Pull request merged**